### PR TITLE
tests: do reset of tests during restore and add checks to validate the fs

### PR DESCRIPTION
**Pull request by sergiocazzolato**
The reset is gonna be done during tests restore
There are a new set of checks which are intended to see if some part of
the filesystem and the system are correctly restored.
Things initial checks are:
. Sytemd status
. Snaps and their status
. Mount points
. Systemd units
. Packages installed

### CODE REVIEW

File: tests/lib/prepare-restore.sh [hunk 1]
```diff
@@ -565,6 +563,145 @@ restore_suite_each() {
         fi
         sleep 1
     done
+
+    # shellcheck source=tests/lib/reset.sh
+    "$TESTSLIB"/reset.sh --reuse-core
+
+    # Reset failing orphan services before checks
+    systemctl reset-failed
```
Comment (mvo5) [inline tests/lib/prepare-restore.sh]: Nice

File: tests/lib/prepare-restore.sh [hunk 2]
```diff
@@ -565,6 +563,145 @@ restore_suite_each() {
...
+get_logs_file() {
+    logs_dir="$RUNTIME_STATE_PATH/logs"
+    logs_id=$(find "$logs_dir" -maxdepth 1 -name '*.journal.log' | wc -l)
+    logs_file=$(echo "${logs_id}_${SPREAD_JOB}" | tr '/' '_' | tr ':' '__')
+
+    echo "${logs_dir}/${logs_file}"
+}
+
+save_pkgs() {
+    file=$1
```
Comment (mvo5) [inline tests/lib/prepare-restore.sh]: Shellcheck would probably complain about missing quotes around $1, i.e. I think this should be `file="$1"`. Or am I missing something? (here and in some more places).

**Comment by stolowski** [comment]
Needs de-conflicting.

**Comment by sergiocazzolato** [comment]
This is closed temporarly

**Pull request closed without merging**